src/c_api/utils.cpp

```diff
@@ -43,3 +48,18 @@ dlaf::common::Ordering char2order(const char order) {
   return order == 'C' or order == 'c' ? dlaf::common::Ordering::ColumnMajor
                                       : dlaf::common::Ordering::RowMajor;
 }
+
+dlaf::comm::CommunicatorGrid& grid_from_context(int dlaf_context) {
+  try {
+    return dlaf_grids.at(dlaf_context);
+  }
+  catch (const std::out_of_range& e) {
+    std::stringstream s;
+    s << "[Error] No DLA-Future grid for context " << dlaf_context << ".\n";
+    s << "[Info]  Make sure you called dlaf_create_grid() or dlaf_create_grid_from_blacs().\n";
```

Comment:
This is major bikeshedding, but should we make this 
```suggestion
    s << "[ERROR] No DLA-Future grid for context " << dlaf_context << ". ";
    s << "Make sure you called dlaf_create_grid() or dlaf_create_grid_from_blacs().\n";
```
?

We're not super consistent about the style of error messages but maybe it's good to not introduce a new style here. We have at the moment:
- No special prefix, i.e. print message and terminate.
- `[ERROR] <message>` and terminate.
- `[{CU,ROC}{SOLVER,BLAS} ERROR] <message>` and terminate.

Reply:
Good point. I think I had SIRIUS' output in mind when I wrote this.

What I was trying to achieve is not to conflate the error message (i.e. there is no grid corresponding to the context), and the suggestion of a possible cause. What about leaving the `\n` in the first line but print the second one without `[ERROR]`? Or phrasing it differently, as a question:

```
[ERROR] No DLA-Future grid for context 12345. Did you forget to call dlaf_create_grid() or dlaf_create_grid_from_blacs()?
```

Although it might sound a bit condescending, I don't know.

Reply:
Not sure how error message are printed, please check asserts and replicate it.

Reply:
That's what I did (although I mistakenly wrote `Error` instead of `ERROR`)

https://github.com/eth-cscs/DLA-Future/blob/c970ce57311d2161714f5461ec018ee724addad0/include/dlaf/common/assert.h#L46

But the question here is how to add the additional information (which is not part of the error, but I think it might be useful for an external user since it points to the most likely cause).

Reply:
Cool, let's go with that then.